src/frontend/src/components/CheckList.tsx

```diff
@@ -42,11 +49,33 @@ const CheckList: React.FC<CheckListProps> = ({ title, headerRight, items }) => {
                 </p>
               }
               checked={check.resolved}
-              onChange={() => handleCheck(idx)}
+              onChange={() => {
+                if (check.resolved) {
+                  setCurrIdx(idx);
+                  setShowConfirm(true);
+                } else {
+                  handleCheck(idx);
+                }
+              }}
             />
           </div>
         ))}
       </Form>
+      {showConfirm ? (
+        <Modal size="lg" centered show={showConfirm} onHide={() => setShowConfirm(false)}>
+          <Modal.Header closeButton>
+            <Modal.Title> Are you sure you want to mark this completed task as NOT completed?</Modal.Title>
+          </Modal.Header>
+          <Modal.Footer className="justify-content-around">
+            <Button onClick={() => handleUncheck(currIdx)} variant="success" type="submit" className="mb-3">
+              Yes
+            </Button>
+            <Button onClick={() => setShowConfirm(false)} variant="danger" className="mb-3">
+              No
+            </Button>
+          </Modal.Footer>
+        </Modal>
```

Comment:
This modal looks bad (would preferably be smaller so the two buttons aren't so far apart for example), I'd say either find a way to make it look better with what we have rn, or just wait to add this until after we finish the mui migration